derp/derp_test.go

```diff
@@ -785,6 +790,72 @@ func TestForwarderRegistration(t *testing.T) {
 	})
 }
 
+type channelFwd struct {
+	id int
+	c  chan []byte
+}
+
+func (f channelFwd) ForwardPacket(_ key.NodePublic, _ key.NodePublic, packet []byte) error {
+	f.c <- packet
+	return nil
+}
+
+func TestMultiForwarder(t *testing.T) {
+	received := 0
+	var wg sync.WaitGroup
+	ch := make(chan []byte)
+	ctx, cancel := context.WithCancel(context.Background())
+	wg.Add(1)
```

Comment:
Let's do `wg.Add(2)` here rather than worrying about whether there are concurrent Done calls with Add (which WaitGroup panics on).

Looks safe, though, because cancel isn't called until late, but I don't want readers to even think about that.

Reply:
Done.